hal/src/thumbv7em/pukcc.rs

```diff
@@ -567,6 +564,7 @@ impl Pukcc {
     ///
     /// All RSA variants up to **RSA4096** (included) will fit into CryptoRAM
     /// and therefore are supported.
+    #[allow(clippy::just_underscores_and_digits)]
```

Comment:
is the issue here the double underscores? If so could we just fix that?

Reply:
Yeah, it complains about tbe double underscores. But also the module is full of super weird syntax [like this](https://github.com/atsamd-rs/atsamd/blob/62f5b959341545ff07b3b03e98b9249a3519ee0b/hal/src/thumbv7em/pukcc.rs#L617) which I don't understand, so I didn't want to touch it and break something.

Reply:
That is really weird...

Reply:
@vccggorski, would you mind explaining what the double underscore variables du throughout the module? If they serve a necessary practical purpose obviously I don't want to touch them, but it there's a way to refactor the module to avoid having to explicitly `#[allow]` the lints that would be great.

Reply:
Sounds good, think I'll just `#[allow]` the lint for the entire module then!

Reply:
Nice :ok_hand: We went through these lints today with @vcchtjader and experimented with alternatives like `_padding` but it doesn't really express every use case of `__` in there plus it obstructs the code more than simple `__`.

Really nice job @jbeaurivage, quite mundane task but it'll help with keeping the repo clean for sure :)